dice-engine/README.md

```diff
@@ -33,17 +33,25 @@ problem in Clojure.
 
 **Dice-roll notation:** `NdX`
 
+**Die map representation**
+
+```clojure
+(def die {:value           1
+          :discarded       false
+          :number_of_faces 3
+          :previous-values [2 2]})
+```
+
 **Dice-roll map representation -**
 
 `3d4`
 
 ```clojure
-(def dice-roll {:numeric_value  8
-                :set            #{4 3 1}
-                :discarded_dice #{}})
+(def dice-roll {:numeric-value 8
+                :dice          [d1 d2 d3]})  ; d1, d2 and d3 are die maps of face value 4, 3 & 1
```

Comment:
We should explain why we have done this change.
Why did we move away from sets to vectors and also removed the `discarded_dice` from the `dice-roll` map.

A commit body can contain this information.

Reply:
Updated the commit message elaborating this.